batch/batch/batch.py

```diff
@@ -1252,9 +1252,14 @@ def image_pull_back_off_reason(container_status):
                               failure_reason="\n".join(image_pull_back_off_reasons))
             return
 
-    if not pod or (pod.status and pod.status.reason == 'Evicted'):
+    if not pod and not app['pod_throttler'].is_queued(job):
```

Comment:
I don't see how we'd ever reach the eviction block of code. If a pod has been evicted then it won't be in the queued jobs and thus always be in the first if block.

Reply:
Cotton wrote this part. I believe his understanding was that we'd receive a k8s event (which this function also handles).

Reply:
erm never mind I misunderstood.

This block is when a pod mysteriously disappears and we don't get an eviction notice. This can happen if we flub somewhere else and don't create the pod or an event is dropped for unknown reasons. In that case, we only want to do all this mark_unscheduled work if the job isn't already in our queue.

Reply:
The eviction block is only for eviction events